clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -864,6 +864,9 @@ else if (req.isInternalRequest && body instanceof ApiVersionsResponse)
                 receive.close();
             }
         }
+
+        // Clear the completed receives once they have been added to ClientResponse returned via poll()
+        this.selector.completedReceives().clear();
```

Comment:
why this clear is needed? shouldn't the latter ones override the previous ones, and the last one will be cleared by the close call?

Reply:
This is to avoid returning the same ByteBuffer to MemoryPool twice. If the close is called but the ByteBuffer will be returned twice to MemoryPool by close in NetworkClient and the close by CompletedReceive in Fetcher